idp/adfs/adfs.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package adfs is an identity provider that authenticates with an ADFS
+// service.
+package adfs
+
+import (
+	oidc "github.com/coreos/go-oidc"
+	"gopkg.in/errgo.v1"
+
+	"github.com/canonical/candid/idp"
+	"github.com/canonical/candid/idp/openid"
+)
+
+func init() {
+	idp.Register("adfs", func(unmarshal func(interface{}) error) (idp.IdentityProvider, error) {
+		var p Params
+		if err := unmarshal(&p); err != nil {
+			return nil, errgo.Notef(err, "cannot unmarshal adfs parameters")
+		}
+		if p.URL == "" {
```

Comment:
nitpick: i'd put this checks in 

func (p *Params) validate() error {
}

Reply:
I'm going to keep[ this as is as we do this in a few places, I might change this a follow-up in a number of IDPs though.